contracts/modules/licensing/LicensingModuleV2.sol

```diff
@@ -0,0 +1,360 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.23;
+
+// external
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { ERC165Checker } from "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol";
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import { IIPAccount } from "../../interfaces/IIPAccount.sol";
+import { IModule } from "../../interfaces/modules/base/IModule.sol";
+import { ILicensingModuleV2 } from "../../interfaces/modules/licensing/ILicensingModuleV2.sol";
+import { IIPAccountRegistry } from "../../interfaces/registries/IIPAccountRegistry.sol";
+import { IDisputeModule } from "../../interfaces/modules/dispute/IDisputeModule.sol";
+import { ILicenseRegistryV2 } from "../../interfaces/registries/ILicenseRegistryV2.sol";
+import { Errors } from "../../lib/Errors.sol";
+import { Licensing } from "../../lib/Licensing.sol";
+import { IPAccountChecker } from "../../lib/registries/IPAccountChecker.sol";
+import { RoyaltyModule } from "../../modules/royalty/RoyaltyModule.sol";
+import { AccessControlled } from "../../access/AccessControlled.sol";
+import { LICENSING_MODULE_KEY } from "../../lib/modules/Module.sol";
+import { BaseModule } from "../BaseModule.sol";
+import { GovernableUpgradeable } from "../../governance/GovernableUpgradeable.sol";
+import { ILicenseTemplate } from "contracts/interfaces/modules/licensing/ILicenseTemplate.sol";
+import { IMintingFeeModule } from "contracts/interfaces/modules/licensing/IMintingFeeModule.sol";
+import { IPAccountStorageOps } from "../../lib/IPAccountStorageOps.sol";
+import { IHookModule } from "../../interfaces/modules/base/IHookModule.sol";
+import { ILicenseNFT } from "../../interfaces/registries/ILicenseNFT.sol";
+
+// TODO: consider disabling operators/approvals on creation
+/// @title Licensing Module
+/// @notice Licensing module is the main entry point for the licensing system. It is responsible for:
+/// - Attaching license terms to IP assets
+/// - Minting license Tokens
+/// - Registering derivatives
+contract LicensingModuleV2 is
```

Comment:
V2 reserved for ugprades in deployed contracts

Reply:
Will remove V2 before merging.